zim/plugins/pageindex/__init__.py

```diff
@@ -474,6 +475,10 @@ def do_row_activated(self, treepath, column):
 		treeiter = model.get_iter(treepath)
 		mytreeiter = model.get_user_data(treeiter)
 
+		if self.last_selected_path == model.get_indexpath(treeiter):
+			return
+		self.last_selected_path = model.get_indexpath(treeiter)
```

Comment:
Use a local variable instead of calling `model.get_indexpath` twice

Reply:
I've replaced get_indexpath with get_selected_path which looks clearer to me.